feat(lib): add `skip_animations` compatibility

Rapsssito opened the pull request:
## Fixes Issue

Fixes #514

## Description

As described in title.

## Check List

Check all the applicable boxes:

- [x] I understand that my contributions needs to pass the checks;
- [x] If I created new functions / methods, I documented them and add type hints;
- [x] If I modified already existing code, I updated the documentation accordingly;
- [x] The title of my pull request is a short description of the requested changes.

jeertmans commented:
Hi @Rapsssito, thanks for your contribution!

Small comments:

1. Did you test your code? It doesn't look like `skip_animations` is passed to Manim (which will thus render the animations, but skip them from Manim Slides's config). I think you need to edit the following
  https://github.com/jeertmans/manim-slides/blob/541b17536005b56061799d91616b803f8f2c6171/manim_slides/slide/manim.py#L107-L118
  and set something like `Scene.next_section(self, *args, skip_animations=base_slide_config.skip_animations, **kwargs)` to make sure the argument is passed to `next_section`.
2. Could you add a test in `tests/test_slides.py::TestSlide`?
3. Could you document your changes in the CHANGELOG?

Many thanks!

Rapsssito commented:
@jeertmans, I did test the code, but I just checked if the slide was indeed not there... My bad. I am sorry about my ignorance, but how can I test it in `tests/test_slides.py::TestSlide`? Is there a way to check for the existence intermediate animation files?

jeertmans commented:
We don't need to go that far in the testing, especially as it would highly depend on where Manim stores partial animation files.

But you could add a test that reads the `PresentationConfig` object (from the generated `.json` file) and checks that slides are indeed missing when `skip_animations=True` is set.

jeertmans commented:
Note, it might be interesting to document this new parameter here: https://github.com/jeertmans/manim-slides/blob/ef282300f1ba9c3492f15570b6e8ff41e7f42f08/manim_slides/slide/base.py#L298-L341

and mention that it is passed to ```
`:meth:`Scene.next_section<manim.scene.scene.Scene.next_section>`,
            or ignored if `manimlib` API is used.```

Rapsssito commented:
I have added the fix and the changelog. However, I am afraid I need your assistance with the tests. I was not able to execute them in my environment with `uv run pytest .\tests\test_slide.py` because I get the error `ModuleNotFoundError: No module named 'manimlib'`. Also, I am not sure how should I access the `PresentationConfig` config from the `TestSlide` class.

> Note, it might be interesting to document this new parameter here:
> 
> https://github.com/jeertmans/manim-slides/blob/ef282300f1ba9c3492f15570b6e8ff41e7f42f08/manim_slides/slide/base.py#L298-L341
> 
> and mention that it is passed to `` `:meth:`Scene.next_section<manim.scene.scene.Scene.next_section>`, or ignored if `manimlib` API is used. ``

I have also added it, but reading the documentation, I would assume it is the same as (I actually did until I discovered it raised an Exception):

```
            :param args:
            Positional arguments passed to
            :meth:`Scene.next_section<manim.scene.scene.Scene.next_section>`,
            or ignored if `manimlib` API is used.
```

jeertmans commented on manim_slides/slide/base.py:
```diff
@@ -299,6 +299,9 @@ def next_slide(
             Positional arguments passed to
             :meth:`Scene.next_section<manim.scene.scene.Scene.next_section>`,
             or ignored if `manimlib` API is used.
+        :param skip_animations:
+            Passed to `:meth:`Scene.next_section<manim.scene.scene.Scene.next_section>`,
+            or ignored if `manimlib` API is used.
```
```suggestion
        :param skip_animations:
            Exclude the next slide from the output.
        
            If `manim` is used, this is also passed to `:meth:`Scene.next_section<manim.scene.scene.Scene.next_section>`,
            which will avoid rendering the corresponding animations.
```

jeertmans replied:
I think this is better documented this way :-)

Rapsssito commented:
In case you miss it, the tests are still missing! I need your help with the implementation.

jeertmans commented:
> In case you missed it, the tests are still missing! I need your help with the implementation.

I didn't miss, I was just waiting for your feedback on that :-)

I will implement the tests myself then ;-)

jeertmans commented:
Thanks for your contribution @Rapsssito!
